src/main/java/org/kuali/kfs/pdp/batch/service/impl/Iso20022FormatExtractor.java

```diff
@@ -1815,16 +1815,31 @@ private String determineFilename(
         }
 
         final Date disbursementDate = extractTypeContext.getDisbursementDate();
-        final SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMdd_HHmmss", Locale.US);
+        final String filename = buildUniqueFilename(directoryName, finalFormattedCheckFilePrefix, disbursementDate, null);
+
+        LOG.debug("determineFilename(...) - : filename={}", filename);
+        return filename;
+    }
+    
+    /* MOD: buildUniqueFilename helper function to ensure file uniqueness to prevent overwriting */
+    private String buildUniqueFilename(String directoryName, String finalFormattedCheckFilePrefix, Date disbursementDate, Integer uniquenessSuffix) {
+        final SimpleDateFormat sdf = new SimpleDateFormat("yyyyMMdd_HHmmssSSS", Locale.US);
+
         final String filename =
                 directoryName
                 + File.separator
                 + finalFormattedCheckFilePrefix
                 + "_"
                 + sdf.format(disbursementDate)
+                + (uniquenessSuffix != null ? "-" + uniquenessSuffix : "")
                 + ".xml";
 
-        LOG.debug("determineFilename(...) - : filename={}", filename);
+        if (new File(filename).exists()) {
+            LOG.info("buildUniqueFilename filename={} already exists, adding uniqueness suffix", filename);
+            int suffixCounter = uniquenessSuffix == null ? 0 : ++uniquenessSuffix;
+            return buildUniqueFilename(directoryName, finalFormattedCheckFilePrefix, disbursementDate, suffixCounter);
```

Comment:
This back-and-forth shifting between primitive and wrapper integers is a little confusing. You may want to consider using just a primitive integer instead, set it 0 or 1 as the initial value, and exclude it from the filename if it's currently at the starter value. It might also be helpful to rename the variable to "fileIndex" or something along those lines.